src/celeritas/em/process/CoulombScatteringProcess.hh

```diff
@@ -0,0 +1,80 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/em/process/CoulombScatteringProcess.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <memory>
+
+#include "celeritas/em/data/WentzelData.hh"
+#include "celeritas/io/ImportParameters.hh"
+#include "celeritas/mat/MaterialParams.hh"
+#include "celeritas/phys/Applicability.hh"
+#include "celeritas/phys/ImportedProcessAdapter.hh"
+#include "celeritas/phys/Process.hh"
+
+namespace celeritas
+{
+
+namespace detail
+{
+//! Coulomb Scattering configuration options
+struct CoulombScatteringOptions
+{
+    //! Nuclear form factor model
+    NuclearFormFactorType form_factor_model{NuclearFormFactorType::exponential};
+
+    //! User defined screening factor
+    real_type screening_factor{1};
+
+    //! Check if the options are valid
+    explicit operator bool() const { return screening_factor > 0; }
+};
```

Comment:
Make these options a nested class inside CoulombScatteringProcess, rather than a detail. (The only reason to avoid nested classes is when the class is templated.)

Reply:
I started off doing this, but I'd like to also pass the options to the model since that's what actually needs them. Including the process header was giving me a lot of cuda compiler errors (I'm purely guessing but I think it's because the process class is only host code while the model will be both host and device and they'll generate conflicting definitions in includes like `MaterialView`). Being able to forward declare the options class in the model file fixes this issue as the process only needs to be included in the .cc file.

I can fully unpack the options and pass them to the model, or I can also create option classes for the model and just copy the data over, but I don't think my attempt at nested classes will work.

Reply:
OK forward declarations are another  good reason not to nest the class 😄. I think you're right about `MaterialView`: I've seen that as well. I think it'd be better to put the options at the model level, since these are really adjusting the model implementation rather than the behavior of the process. Or are they reusable across multiple models in practice? If they're specific to the Wenzel model, call them `WenzelModelOptions` and put them in the `WenzelModel` header (and in the main namespace, not `detail`).